modules/andes-core/broker/src/main/java/org/wso2/andes/server/queue/SimpleAMQQueue.java

```diff
@@ -522,8 +523,34 @@ public void setDeleteOnNoConsumers(boolean b)
         _deleteOnNoConsumers = b;
     }
 
-    public void addBinding(final Binding binding)
-    {
+    /**
+     * Add binding for the queue. Check if there are more than 1 binding other than the default
+     * binding. If so thrown an exception.
+     * @param binding  binding to be added
+     * @throws AndesException
+     */
+    public void addBinding(final Binding binding) throws AndesException {
+        if (_bindings.size() == 0) {
+            addBindingToQueue(binding);
+        } else if (_bindings.size() == 1) {
+            if (_bindings.get(0).getExchange().getNameShortString().equals(ExchangeDefaults
+                    .DEFAULT_EXCHANGE_NAME.toString())) {
+                addBindingToQueue(binding);
+            } else {
+                _logger.error("Binding already exists for the queue. Cannot be permitted");
+                throw new AndesException("Binding already exists for the queue");
```

Comment:
Yes. Because at the end of the day, this error is thrown only to client side. at MB side this error is not printed. Therefore error is printed here.